pkg/types/azure/validation/machinepool.go

```diff
@@ -12,5 +16,40 @@ func ValidateMachinePool(p *azure.MachinePool, fldPath *field.Path) field.ErrorL
 	if p.OSDisk.DiskSizeGB < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("diskSizeGB"), p.OSDisk.DiskSizeGB, "Storage DiskSizeGB must be positive"))
 	}
+
+	if p.OSDisk.DiskType != "" {
+		diskTypes := sets.NewString("Standard_LRS", "StandardSSD_LRS", "Premium_LRS") // "UltraSSD_LRS"
```

Comment:
nit: Personally I would rather see one of two options:
1. a more substantial comment: e.g. "Adding UltraSSD_LRS depends on..."
2. moving this comment to be part of the data: 
```go
diskTypes := sets.NewString("Standard_LRS", 
                                              // "UltraSSD_LRS", // needs azure terraform provider xx
                                               "StandardSSD_LRS", 
                                               "Premium_LRS") 
``` 
this is my personal preference and not sure if others share the same opinion.
note: I consider this as "data" and am ok leaving in commented data. we should not leave in commented code.